proto/akash/deployment/deployment.proto

```diff
@@ -87,9 +87,23 @@ message Deployment {
         (gogoproto.jsontag)    = "id",
         (gogoproto.moretags)   = "yaml:\"id\""
     ];
-    uint32 state = 2 [
-        (gogoproto.customtype) = "DeploymentState",
-        (gogoproto.nullable)   = false,
+
+    enum State {
+        option (gogoproto.goproto_enum_prefix) = false;
+
+        // Prefix should start with 0 in enum. So declaring dummy state
+        all = 0 [
+            (gogoproto.enumvalue_customname) = "DeploymentStateDefault"
+        ];
```

Comment:
Can you name these `invalid`, `"*Invalid"`?  Leaving out the zero value was intentional to make it so that we had to be explicit about things.